app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/tokens/RadioOptionTokens.kt

```diff
@@ -0,0 +1,75 @@
+package com.hedvig.android.design.system.hedvig.tokens
+
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.hedvig.android.design.system.hedvig.RadioOptionDefaults.RadioOptionStyle
+import com.hedvig.android.design.system.hedvig.RadioOptionDefaults.RadioOptionStyle.Label
+
+internal object RadioOptionColorTokens {
+  val ContainerColor = ColorSchemeKeyTokens.SurfacePrimary
+  val OptionTextColor = ColorSchemeKeyTokens.TextPrimary
+  val LabelTextColor = ColorSchemeKeyTokens.TextSecondaryTranslucent
+  val DisabledOptionTextColor = ColorSchemeKeyTokens.TextDisabledTranslucent
+  val DisabledLabelTextColor = ColorSchemeKeyTokens.TextDisabledTranslucent
+  val ChosenIndicatorColor = ColorSchemeKeyTokens.SignalGreenElement
+  val NotChosenIndicatorColor = ColorSchemeKeyTokens.BorderSecondary
+  val DisabledIndicatorColor = ColorSchemeKeyTokens.FillDisabledTransparent
+}
+
+internal abstract class SizeRadioOptionTokens {
+  protected abstract val TopPadding: Dp
+  protected abstract val BottomPadding: Dp
+  protected abstract val LabelTopPadding: Dp
+  protected abstract val LabelBottomPadding: Dp
+  abstract val OptionTextFont: TypographyKeyTokens
+  abstract val LabelTextFont: TypographyKeyTokens
+  abstract val HorizontalPadding: Dp
+  abstract val ContainerShape: ShapeKeyTokens
+
+  fun topPadding(style: RadioOptionStyle): Dp {
+    return when (style) {
+      is Label -> LabelTopPadding
+      else -> TopPadding
+    }
+  }
+
+  fun bottomPadding(style: RadioOptionStyle): Dp {
+    return when (style) {
+      is Label -> LabelBottomPadding
+      else -> BottomPadding
+    }
+  }
```

Comment:
I have a hunch that the tokens here shouldn't really need to know anything about the style and should hopefully just be raw tokens ready to be used in whatever way.
At the same time we don't want to repeat ourselves over at the place where we call these.
I think I had the same scenario over here https://github.com/HedvigInsurance/android/blob/ee04408d62eba72d61de7391b832fcb4d2513fab/app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/HedvigTextField.kt#L347-L361 where I added this piece of logic inside the style itself, and the implementation of each size just knew about both LabelTopPadding and TopPadding and that function in there picked the right one for all three sizes.